### ISSUE TITLE
Model fitting / parameter selection

### ISSUE THREAD

**Issue by charlesdgburns** [charlesdgburns/NM_TinyRNN#7]
Re-write the training loop for model parameter selection.

The most robust method would be the inner-outer-loops used in [Li Ji-An et al. (2025)](https://www.nature.com/articles/s41586-025-09142-4).

However, if we want to fit data on smaller samples (for example on subsets of data early in learning versus later in learning), it might be fruitful to use the Goldfish training method ([Hans et al,. 2024](https://arxiv.org/abs/2406.10209))

**Comment by charlesdgburns** [comment]
Thought here: ought to test batch_size influence. I think following the 'tricks of the trade' book, a small batch size should be the fastest training in real time [(Orr et al., - there should be a more recent verison than this)](https://link.springer.com/book/10.1007/3-540-49430-8).

Noticed that due to recurrence, at least with a small batch size it is faster on a CPU than GPU.

___

outputs: rather than outputting each model weights, output the average validation loss, N_epochs, and training losses for each model - then save only the weights for the best model. Eval score is also saved for the best model, for comparison across model-types (or same model-type across subjects).

We can get away with this by making training deterministic-  unless we should do multiple instantiations per set of parameters (arguably not required due to stability of Adam) - then with a set seed the same model weights can be retrieved by rerunning a specific hyperparameters.

**Comment by charlesdgburns** [comment]
Seems to be an issue with the training.py script -at least when run in google COLAB.

A few features / bugfixes:

- remove the forced_choice trial preictions from val_pred_loss
- Consider running inner + outer loops over sessions, since ideally the session-by-session activations (for later NM regression) are excluded from the training of the model that we fit (but this might not be necessary for testing).

**Comment by charlesdgburns** [comment]
Fixed the bug (had to do with early_stop_loss not working properly).
Removed forced_choice predictions from val_pred_loss.

instead of running the crazy 90 models (although this might just be a necessary evil), we could just do a mean over 10 random seeds per subject.

**Comment by charlesdgburns** [comment]
Good to know: the majority of the variance comes from different selection of training, validation, and evaluation trials.

However, a bit of variability still is due to the initialisation (!). Curious how one of the 1D hidden unit models seems to perform as well as a 2D GRU. Worth taking an extra look (!)

<img width="586" height="294" alt="Image" src="https://github.com/user-attachments/assets/3f00267d-b600-4459-afb3-0a8365803a08" />
